src/edit.js

```diff
@@ -77,6 +77,15 @@ function Edit( props ) {
 		[ clientId ]
 	);
 
+	const [ useCustomUrl, setUseCustomUrl ] = useState( false );
```

Comment:
Thanks for working on this! 

Because the hook is called after a return, that makes it conditional and violates the usage rules of a hook.
Can you move `useState` to before line 58?